### PR TITLE
Add documentation comments section to coding guidelines

### PR DESCRIPTION
**Pull request by xtqqczze**
* Add new documentation comments section
* Retarget doc comments link to conceptual article

Follow-up to #14314.

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: docs/dev-process/coding-guidelines.md [hunk 1]
```diff
@@ -86,8 +86,13 @@ We also run the [.NET code formatter tool](https://github.com/dotnet/codeformatt
 
 * Make sure the added/updated comments are meaningful, accurate and easy to understand.
 
-* Public members must use [doc comments](https://docs.microsoft.com/dotnet/csharp/programming-guide/xmldoc/).
-  Internal and private members may use doc comments but it is not required.
+### Documentation comments
+
+* Create documentation using [XML documentation comments](https://docs.microsoft.com/dotnet/csharp/codedoc) so that Visual Studio and other IDEs can use IntelliSense to show quick information about types or members.
+
+* Publicly visible types and their members must be documented.
```
Comment (iSazonov) [inline docs/dev-process/coding-guidelines.md:93]:
```suggestion
* Publicly visible types and their members must be documented.
  Internal and private members may use doc comments but it is not required.
```

Comment (xtqqczze) [reply docs/dev-process/coding-guidelines.md:93]: This is implicit in the previous bullet point, do you think we need we need this sentence?

**Comment by adityapatwardhan** [comment]
I will merge this after the CI finishes,

**Comment by xtqqczze** [comment]
@adityapatwardhan You could merge now.

`PowerShell-CI-static-analysis` failures are unrelated (`markdown-link` failures).

**Comment by adityapatwardhan** [comment]
@xtqqczze Thank you for your contribution!

**Pull request merged**